### ISSUE TITLE
Hoverlabel overlaps each other

### ISSUE THREAD

**Issue by amqadmiakur8** [plotly/plotly.js#6917]
Hello,

I found an issue causing hover labels to overlap each other.
<img width="535" alt="image" src="https://github.com/plotly/plotly.js/assets/65157198/8f274b06-244f-4bbc-bc47-37a20b31e1c5">
It can look quite bad in some cases:
<img width="617" alt="image" src="https://github.com/plotly/plotly.js/assets/65157198/d175f5a0-9e51-40de-9a9e-66aef75e0e65">

Here is a codepen to reproduce it
https://codepen.io/Amine-Mqadmi/pen/JjVGpym

I think it is has been introduced in 2.19.0, probably in this PR: https://github.com/plotly/plotly.js/pull/6442
You can see in the discussion @alexcjohnson noticing it, but it seems it has not been actually fixed in the end.

**Comment by rmasachs** [comment]
Hello, 
Is there any update on this issue?

**Comment by mbant** [comment]
@archmoj @ndrezn for visibility on the right people maybe? 
Do you know if this is something that you'll ever be interested in having a look at? Currently blocking us from upgrading to 2.19.0 and beyond.

It seems that mixing `bar` and `scatter` traces the overlap the problem reveal itself and you can construct quite weird examples where labels not only overlap but also 'disappear'!

<img width="647" alt="Screenshot 2024-04-09 at 18 23 36" src="https://github.com/plotly/plotly.js/assets/42407285/72df562a-7e5a-480b-aa04-fc93ca02bcbc">


Reverting commit https://github.com/plotly/plotly.js/commit/2b4584dfe54598db03e8d9724a766789293b4f69 indeed fixes the issue (in particular just reverting [this line](https://github.com/plotly/plotly.js/commit/2b4584dfe54598db03e8d9724a766789293b4f69#diff-754e3dde48df955de63f6c76eb1c43fed92a3931b9d7f51035a2d400029c8063L1521) ) but of course re-introduces potential overlap of labels with axis labels.

<img width="659" alt="Screenshot 2024-04-09 at 18 24 34" src="https://github.com/plotly/plotly.js/assets/42407285/9ecc4f71-0fab-4983-9a31-710143f58494">


<details>
  <summary>Here's the JSON I used for testing in the dashboard</summary>

```json
{
  "data": [
    {
      "type": "scatter",
      "y": [0, 1, 2, 3],
      "x": ["CAT 1", "CAT 2", "CAT 3", "CAT 4"]
    },
    {
      "type": "bar",
      "y": [0, 1, 2, 3],
      "x": ["CAT 1", "CAT 2", "CAT 3", "CAT 4"]
    },
    {
      "type": "scatter",
      "y": [0, 1, 2, 3],
      "x": ["CAT 1", "CAT 2", "CAT 3", "CAT 4"]
    },
    {
      "type": "bar",
      "y": [0, 1, 2, 3],
      "x": ["CAT 1", "CAT 2", "CAT 3", "CAT 4"]
    }
  ],
  "layout": {
    "width": 600,
    "height": 400,
    "hovermode": "x"
  }
}
```

</details>

**Comment by mbant** [comment]
@archmoj / @alexcjohnson attempting a small fix (one-liner) for this in https://github.com/plotly/plotly.js/pull/6954 if you're open for contributions! 
CI is passing now, apologies for the couple extra commits I had a different esline config in my IDE that generated a couple of issues ...

**Comment by archmoj** [comment]
Fixed by #6954.
